UefiPayloadPkg/UefiPayloadEntry/UefiPayloadEntry.c

```diff
@@ -23,6 +23,9 @@ PayloadEntry (
   PHYSICAL_ADDRESS              DxeCoreEntryPoint;
   EFI_HOB_HANDOFF_INFO_TABLE    *HandoffHobTable;
   EFI_PEI_HOB_POINTERS          Hob;
+  UINTN                         ImageBase;

```

Comment:
Can you define "EFI_FIRMWARE_VOLUME_HEADER *PayloadFv" here and assign the HOB value to PayloadFv?
And change the first parameter of LoadDxeCore() to "EFI_FIRMWARE_VOLUME_HEADER *PayloadFv".
This is to avoid the additional typecast in LoadDxeCore().